massa-execution/src/worker.rs

```diff
@@ -287,9 +304,52 @@ impl ExecutionWorker {
                     address,
                 });
             }
-            ExecutionCommand::GetSCOutputEventBySlotRange { .. } => todo!(),
-            ExecutionCommand::GetSCOutputEventByCaller { .. } => todo!(),
-            ExecutionCommand::GetSCOutputEventBySCAddress { .. } => todo!(),
+            ExecutionCommand::GetSCOutputEventBySlotRange {
+                start,
+                end,
+                response_tx,
+            } => todo!(),
+            ExecutionCommand::GetSCOutputEventByCaller {
+                caller_address,
+                response_tx,
+            } => todo!(),
+            ExecutionCommand::GetSCOutputEventBySCAddress {
+                sc_address,
+                response_tx,
+            } => todo!(),
+            ExecutionCommand::GetSCELedgerForAddresses {
+                response_tx,
+                addresses,
+            } => {
+                let ledger = &self
```

Comment:
Instead of sharing the execution context between the worker and context, you can use a `ExecutionRequest`, like was done in https://github.com/massalabs/massa/pull/2056/

Reply:
Thanks !

Reply:
It's much cleaner that way :)